### PR TITLE
Add docs on how to use containers with slurm

### PR DESCRIPTION
**Pull request by boehmseb**

### CODE REVIEW

File: docs/source/vara-ts/slurm.rst [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            PATH:
+            - /scratch/<user>/varats/tools/VaRA/bin/
+            HOME: /scratch/<user>/
+
+   - configure slurm related parameters:
+
+   .. code-block:: yaml
+
+      jobs: '10' # TODO: find good default
+      parallel_processes: '4' # TODO: find good default
```
Comment (vulder) [inline docs/source/vara-ts/slurm.rst]:
```suggestion
      parallel_processes: '1'
```

File: docs/source/vara-ts/slurm.rst [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+          partition:
+              value: anywhere # or name of the cluster to run on
+          timelimit:
+              value: 'hh:mm:ss'
+
+   - set vara related options:
+
+   .. code-block:: yaml
+
+      vara:
```
Comment (vulder) [inline docs/source/vara-ts/slurm.rst]:
```suggestion
      varats:
```

File: varats/varats/tools/driver_container.py [hunk 3]
```diff
@@ -4,129 +4,188 @@
...
-    elif args['subcommand'] == 'select-tool':
-        __select_research_tool()
+    click.echo("Building base images. This could take a while...")
+    # TODO: clean exported images if present
+    __build_images(images, True, debug)
+    click.echo("Done.")
+
+    click.echo(
+        "Run `benchbuild slurm -E <report_type> <project> -- "
+        "container run --import` inside the `benchbuild` directory"
```
Comment (boehmseb) [inline varats/varats/tools/driver_container.py]:
```suggestion
        "container run --import` inside the `benchbuild` directory "
```

File: varats/varats/tools/templates/slurm_container.sh.inc [hunk 4]
```diff
@@ -0,0 +1,87 @@
+{% raw %}
+#!/bin/bash
```
Comment (boehmseb) [inline varats/varats/tools/templates/slurm_container.sh.inc]:
```suggestion
{% raw %}#!/bin/bash
```

File: varats/varats/tools/driver_container.py [hunk 5]
```diff
@@ -4,129 +4,188 @@
...
+    type=click.Path(),
+    prompt="What should be the base directory on slurm nodes?",
+    help="Base directory on slurm nodes. \n"
+    "Must be creatable and writeable on every slurm node and your local "
+    "machine and must not be on a NFS."
+)
+@click.option(
+    "--export-dir",
+    type=click.Path(),
+    prompt="Where should base images be exported to?",
```
Comment (vulder) [inline varats/varats/tools/driver_container.py]:
```suggestion
    prompt="Where should base images be exported to for storage?",
```

Comment (vulder) [reply varats/varats/tools/driver_container.py]: Maybe, we should also hint the config option here two?

**Review by vulder — CHANGES_REQUESTED** [review]

File: docs/source/vara-ts/slurm.rst [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+        value:
+            PATH:
+            - /scratch/<user>/varats/tools/VaRA/bin/
+            HOME: /scratch/<user>/
+
+   - configure slurm related parameters:
+
+   .. code-block:: yaml
+
+      jobs: '10' # TODO: find good default
```
Comment (vulder) [inline docs/source/vara-ts/slurm.rst]:
```suggestion
      jobs: '20'
```
We have 20 cpus at kine so maybe 20?

File: varats/varats/tools/driver_container.py (hunk 5 above)
Comment (boehmseb) [reply varats/varats/tools/driver_container.py]: Maybe only in the docs? This tool is more intended for users that are not that familiar with the config.

**Review by vulder — APPROVED** [review]

**Review by Sinerum — APPROVED** [review]
Looks good to me

**Pull request merged**